### PR TITLE
feat: feed approved example queries into training and Explore templates (#192)

### LINKED ISSUE #192: feat: feed approved example queries into training pipeline and surface as templates
## Problem

The `/guide` page (`ui-nextjs/app/guide/page.tsx`, Section 3) tells contributors:

> Example queries serve two purposes: they become training data for the fine-tuned model, and they provide ready-made templates that help new users understand what the platform can answer.

Neither is wired up. Approved `example_queries` rows sit in the database and are never read by:

- **Training pipeline** — `scripts/training/generate_training_pairs.py` pulls from its own seed corpus; it has no hook to include staff-contributed queries
- **UI templates** — the research page (`ui-nextjs/app/page.tsx`) and `/explore` (`ui-nextjs/app/explore/page.tsx`) don't surface approved examples as starter templates

## Goal

Two independent deliverables (split into sub-tasks if needed):

### A. Training data integration

1. Extend `scripts/training/generate_training_pairs.py` to read approved rows from `example_queries` where a curated answer is present
2. Produce training pairs in the existing D4BL format (system prompt + user query + assistant JSON)
3. Document the flow so staff see their contributions flow into the next training run

### B. Template surface in UI

1. Add a "Community examples" row on the research page or explore page
2. Pull from approved `example_queries` via a new API endpoint
3. Clicking a template pre-fills the query input

## Acceptance criteria

- [ ] `generate_training_pairs.py` has a flag (e.g. `--include-staff-queries`) that merges approved example queries into the training set
- [ ] A new read-only API endpoint `GET /api/example-queries` returns approved queries (paginated)
- [ ] Research UI shows a handful of curated examples sourced from the `example_queries` table
- [ ] Guide copy (Section 3) accurately describes post-approval behavior

## Design notes

- Training inclusion is batch (next training run), not realtime — be explicit about that in the guide
- Consider whether templates need admin curation separate from approval (e.g. a `is_featured` flag on `example_queries`) to avoid a growing unfiltered list
- Deferred from PR #189 (v2 work)

## Related

- Parent guide page: `ui-nextjs/app/guide/page.tsx` Section 3
- Training pipeline: `scripts/training/generate_training_pairs.py`

### PR DESCRIPTION
**Pull request by William-Hill**
## Summary

This change completes the post-approval path for staff-submitted example queries, consistent with approved documents and approved data-source uploads. Queries that reach `approved` status are pulled from `example_queries` (via `uploads`) into the training pipeline and listed in Explore as reusable NL templates.

## What changed

- **Corpus extraction** adds an `approved_example_queries` source so pretraining JSONL can include real question phrasing. Extraction skips that source cleanly when the upload tables are not present.
- **Query-parser distillation** prepends approved questions (teacher uses community style), then fills the remainder from synthetic seeds. **Query-parser v2** injects approved text into the intersectional batch when the queue is non-empty.
- **API** exposes `GET /api/explore/example-query-templates` for authenticated clients (same pattern as other Explore routes).
- **Explore UI** loads those templates and shows short chips above the conversational query bar; choosing a chip fills the input.

## Testing

Targeted pytest runs covered the new merge helpers, template rendering, API handler wiring, and related Explore and training tests (including `test_extract_corpus`).

Closes #192

Made with [Cursor](https://cursor.com)

## Summary by CodeRabbit

* **New Features**
  * "Try a contributor example" section in Explore with approved example query suggestions.
  * API endpoint to list approved example query templates.
  * Training pipeline opt-in flag to include approved staff example queries with deterministic staging/resume behavior.
  * New renderer to format example-query passages for training/UI.

* **Bug Fixes**
  * Missing or unavailable example-query data now logs a warning and returns empty results instead of failing.

* **Tests**
  * Added tests for the API, rendering, training pair merging/injection, CLI flag, and UI integration.

### CODE REVIEW

File: src/d4bl/app/api.py [hunk 1]
```diff
@@ -1870,6 +1870,34 @@ async def list_staff_upload_datasets(
...
+    rows = result.mappings().all()
+    return [
+        {
+            "id": str(r["id"]),
+            "query_text": r["query_text"],
+            "description": r["description"],
+            "summary_format": r["summary_format"],
+        }
+        for r in rows
+    ]
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply src/d4bl/app/api.py:1905]: Wrapped the DB query in try/except: on any failure we log with exc_info and return an empty list so Explore matches the optional-templates contract.

File: scripts/training/extract_corpus.py [hunk 2]
```diff
@@ -172,7 +184,17 @@ def main(tables: list[str] | None = None, max_per_table: int = MAX_PASSAGES_PER_
...
+            try:
+                passages = extract_table(conn, table, max_per_table)
+            except Exception as exc:  # noqa: BLE001
+                if table == "approved_example_queries":
+                    logger.warning(
+                        "Skipping approved example queries (tables may be absent): %s",
+                        exc,
+                    )
+                    passages = []
+                else:
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply scripts/training/extract_corpus.py]: Replaced the broad except with psycopg2.errors.UndefinedTable only for approved_example_queries; all other errors propagate.

File: ui-nextjs/components/explore/ExploreQueryBar.tsx [hunk 3]
```diff
@@ -32,6 +39,29 @@ export default function ExploreQueryBar({
...
+      } catch {
+        /* ignore — templates are optional */
+      }
+    }
+
+    void loadTemplates();
+    return () => {
+      cancelled = true;
+    };
+  }, [getHeaders]);
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/explore/ExploreQueryBar.tsx]: Template fetch now runs once on mount with an exhaustive-deps eslint disable; avoids refetch loops if getHeaders identity changes each render.

File: scripts/training/extract_corpus.py [hunk 4]
```diff
@@ -172,7 +184,17 @@ def main(tables: list[str] | None = None, max_per_table: int = MAX_PASSAGES_PER_
...
         for table in target_tables:
             logger.info("Extracting %s (max %d rows)…", table, max_per_table)
-            passages = extract_table(conn, table, max_per_table)
+            try:
+                passages = extract_table(conn, table, max_per_table)
+            except Exception as exc:  # noqa: BLE001
+                if table == "approved_example_queries":
+                    logger.warning(
+                        "Skipping approved example queries (tables may be absent): %s",
+                        exc,
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply scripts/training/extract_corpus.py]: Same as the other extract_corpus thread: only UndefinedTable is swallowed for approved_example_queries.

File: scripts/training/generate_training_pairs.py [hunk 5]
```diff
@@ -849,6 +855,24 @@ def generate_query_parser_pairs_v2(
...
+            if (text := (r.get("query_text") or "").strip())
+        ),
+    )
+    if staff_for_v2:
+        print(
+            f"[{task_name}] {len(staff_for_v2)} approved staff queries available "
+            "for intersectional batch injection",
+            flush=True,
+        )
+
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply scripts/training/generate_training_pairs.py]: Added --no-include-approved-example-queries to opt out. For resume stability, approved-query rows are written to sidecar JSON on fresh runs and reloaded on --resume; v2 intersectional queue uses the same pattern. Pushed in f3e7aa61.

**Comment by William-Hill** [comment]
Review feedback addressed in `f3e7aa61`:

- **example-query-templates API**: try/except around the query; log and return `[]` on failure so the UI always gets a stable contract.
- **extract_corpus**: only `psycopg2.errors.UndefinedTable` is skipped for `approved_example_queries`; other failures still fail the run.
- **ExploreQueryBar**: template fetch runs once on mount (eslint disable for exhaustive-deps) to avoid refetch loops.
- **generate_training_pairs**: `--no-include-approved-example-queries` to opt out; staff-approved rows are snapshotted to JSON beside checkpoints on fresh runs and reloaded on `--resume` so indices stay aligned (v2 uses the same pattern for the intersectional deque).

Targeted pytest slice for explore + training modules: 60 passed locally.

File: ui-nextjs/components/explore/ExploreQueryBar.tsx [hunk 6]
```diff
@@ -32,6 +39,31 @@ export default function ExploreQueryBar({
...
+      }
+    }
+
+    void loadTemplates();
+    return () => {
+      cancelled = true;
+    };
+    // Intentionally once on mount: getHeaders identity can change each render and would refetch.
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
```
[the remark this answers is not in the document]
Comment (William-Hill) [reply ui-nextjs/components/explore/ExploreQueryBar.tsx]: Template fetch now waits for `session.access_token`, then runs with `[session?.access_token, getHeaders]` deps so the request retries after auth hydrates (no eslint-disable).

**Comment by William-Hill** [comment]
Follow-up from CodeRabbit (commit `ee0088a8`):

- **ExploreQueryBar**: template fetch runs only after `session.access_token` exists, with effect deps `[session?.access_token, getHeaders]` so hydration no longer yields a permanent empty state.
- **Staff merge CLI**: positive flag `--include-approved-example-queries` (default **off**). `scripts/run_training_pipeline.py` passes `include_approved_example_queries=True` for distill so the bundled pipeline still picks up approved queries (#192).
- **Resume + staff**: if that flag is on and `--resume` is used, a missing or unreadable `.query_parser_staff_snapshot.json` / `.query_parser_v2_staff_snapshot.json` now raises **RuntimeError** instead of silently reordering.
